File: threads/src/main/java/im/threads/internal/database/DBHelper.java
package im.threads.internal.database;

import androidx.annotation.NonNull;

import java.util.List;

import im.threads.business.models.ChatItem;
import im.threads.internal.model.ConsultInfo;
import im.threads.internal.model.ConsultPhrase;
import im.threads.internal.model.FileDescription;
import im.threads.internal.model.MessageState;
import im.threads.internal.model.Survey;
import im.threads.internal.model.UserPhrase;

public interface DBHelper {

    void cleanDatabase();

    @NonNull
    List<ChatItem> getChatItems(int offset, int limit);

    ChatItem getChatItem(String messageUuid);

    void putChatItems(final List<ChatItem> items);

    boolean putChatItem(ChatItem chatItem);

    List<FileDescription> getAllFileDescriptions();

    void updateFileDescription(@NonNull FileDescription fileDescription);

    ConsultInfo getLastConsultInfo(@NonNull String id);

    List<UserPhrase> getUnsendUserPhrase(int count);

    void setUserPhraseStateByProviderId(String providerId, MessageState messageState);

    ConsultPhrase getLastConsultPhrase();

    int setAllConsultMessagesWereRead();

    void setMessageWasRead(String providerId);

    Survey getSurvey(long sendingId);

    int setNotSentSurveyDisplayMessageToFalse();

    int setOldRequestResolveThreadDisplayMessageToFalse();

    int getMessagesCount();

    int getUnreadMessagesCount();

    List<String> getUnreadMessagesUuid();

}

File: threads/src/main/java/im/threads/internal/secureDatabase/DBHelper.kt
package im.threads.internal.secureDatabase

import im.threads.business.models.ChatItem
import im.threads.internal.model.ConsultInfo
import im.threads.internal.model.ConsultPhrase
import im.threads.internal.model.FileDescription
import im.threads.internal.model.MessageState
import im.threads.internal.model.Survey
import im.threads.internal.model.UserPhrase

interface DBHelper {

    fun cleanDatabase()
    fun getChatItems(offset: Int, limit: Int): List<ChatItem?>
    fun getChatItem(messageUuid: String?): ChatItem?
    fun putChatItems(items: List<ChatItem?>?)
    fun putChatItem(chatItem: ChatItem?): Boolean

    fun getAllFileDescriptions(): List<FileDescription?>?
    fun putFileDescriptions(fileDescriptions: List<FileDescription?>)
    fun updateFileDescription(fileDescription: FileDescription)

    fun getLastConsultInfo(id: String): ConsultInfo?
    fun getUnsendUserPhrase(count: Int): List<UserPhrase?>?
    fun setUserPhraseStateByProviderId(providerId: String?, messageState: MessageState?)
    fun getLastConsultPhrase(): ConsultPhrase?

    fun setAllConsultMessagesWereRead(): Int
    fun setAllConsultMessagesWereReadWithThreadId(threadId: Long?): Int
    fun setMessageWasRead(providerId: String)
    fun getSurvey(sendingId: Long): Survey?
    fun setNotSentSurveyDisplayMessageToFalse(): Int
    fun setOldRequestResolveThreadDisplayMessageToFalse(): Int
    fun getMessagesCount(): Int
    fun getUnreadMessagesCount(): Int
    fun getUnreadMessagesUuid(): List<String?>?
}
